### PR TITLE
luminous: crypto: don't use PK11_ImportSymKey() in FIPS mode

### PR DESCRIPTION
**Pull request by rzarzynski**
This fix is dedicated to Luminous. In master we're switching to OpenSSL.

Fixes: http://tracker.ceph.com/issues/38843

### CODE REVIEW

**Comment by smithfarm** [comment]
@rzarzynski It would be nice if the explanation "This fix is dedicated to Luminous. In master we're switching to OpenSSL." could be included in the commit messages themselves. Otherwise, folks examining the git history will likely miss it.

**Comment by neha-ojha** [comment]
http://pulpito.ceph.com/nojha-2019-03-26_20:35:56-rados-wip-nss-importsymkey-in-fips-luminous-distro-basic-smithi/

File: src/common/ceph_crypto.cc [hunk 1]
```diff
@@ -44,6 +45,119 @@ static uint32_t crypto_refs = 0;
...
+    return ret_key;
+  }
+
+  ceph_assert_always(wincx == nullptr);
+
+  struct {
+    unsigned char data[256];
+    int len;
+  } wrapped_key;
+  ceph_assert_always(raw_key->len <= sizeof(wrapped_key.data));
```
Comment (rzarzynski) [inline src/common/ceph_crypto.cc]: Refactoring this.

**Comment by yuriw** [comment]
[wip-yuri4-testing-2019-03-27-2009-luminous](https://trello.com/c/wBvzoctn/567-wip-yuri4-testing-2019-03-27-2009-luminous)

**Pull request merged**